crates/dt-workspace/src/config/mod.rs

```diff
@@ -0,0 +1,80 @@
+use std::path::PathBuf;
+
+#[cfg(feature = "cli")]
+use cli_config::CliConfig;
+use env_config::EnvConfig;
+use toml_config::TomlConfig;
+
+#[cfg(feature = "cli")]
+pub mod cli_config;
+pub mod env_config;
+pub mod toml_config;
+
+macro_rules! config_field {
+  ($start:ident, $($fallback:ident),+; $field:ident) => {
+      $start.and_then(|value| value.$field)
+        $(
+        .or_else(|| config_field!($fallback; $field))
+        )+
+  };
+  ($struct:ident; $field:ident) => {
+    $struct.and_then(|value| value.$field)
+  }
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct CombinedConfig {
+    include_paths: Vec<PathBuf>,
+}
+
+impl CombinedConfig {
+    #[must_use]
+    pub fn merge(
+        toml: Option<TomlConfig>,
+        #[cfg(feature = "cli")] cli: Option<CliConfig>,
+        env: Option<EnvConfig>,
+    ) -> Self {
+        Self {
+            #[cfg(feature = "cli")]
+            include_paths: config_field!(cli, env, toml; include_paths).unwrap_or_default(),
+            #[cfg(not(feature = "cli"))]
+            include_paths: config_field!(env, toml; include_paths).unwrap_or_default(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn merge() {
+        let toml = TomlConfig {
+            include_paths: Some(vec!["toml".into()]),
+        };
+        #[cfg(feature = "cli")]
+        let cli = CliConfig {
+            include_paths: Some(vec!["cli".into()]),
+        };
+        let env = EnvConfig {
+            include_paths: Some(vec!["env".into()]),
+        };
+
+        assert_eq!(
+            CombinedConfig {
+                include_paths: vec![
+                    #[cfg(feature = "cli")]
+                    "cli".into(),
+                    #[cfg(not(feature = "cli"))]
+                    "env".into(),
```

Comment:
```suggestion
                    if cfg!(feature = "cli") {
                      "cli".to_owned()
                    } else {
                      "env".to_owned()
                    },
```

https://doc.rust-lang.org/std/macro.cfg.html

Note the use of `to_owned` instead of `into`. Could you apply this in other places with `&str` -> `String`? Some say it's better for compile times, but for me it just clearly shows that there is a clone in the code.

Reply:
`into` used for `PathBuf::from` or `Path::new`, should tests be wrapped with these?

Reply:
> ```rs
> vec!["linux".into(), "linux-old".into()]
> ```

Those are fine for now

Reply:
`if cfg!(feature = "cli") { "cli" } else { "env" }.into()` looks better than `PathBuf::from(if cfg!(feature = "cli") { "cli" } else { "env" })` IMO

Reply:
Oh right I thought that was using `into` for `&str` to `String`. Use the one you suggested.